src/System.Collections/src/System/Collections/Generic/CollectionExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace System.Collections.Generic
+{
+    public static class CollectionExtensions
+    {
+        public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key)
```

Comment:
Linq's extension methods throw `ArgumentNullException` when the source is null. I think we should do the same for these extension methods, throwing `ArgumentNullException` if the dictionary is null.